pinot-common/src/main/java/org/apache/pinot/common/function/scalar/DataTypeConversionFunctions.java

```diff
@@ -32,6 +33,23 @@ public class DataTypeConversionFunctions {
   private DataTypeConversionFunctions() {
   }
 
+  @ScalarFunction
+  public static Object cast(Object value, String targetType) {
+    if (value == null) {
+      return null;
+    }
+    try {
+      Class<?> clazz = value.getClass();
+      boolean isMultiValue = clazz.isArray();
+      PinotDataType sourceType = isMultiValue
+          ? PinotDataType.getMultiValueType(clazz)
+          : PinotDataType.getSingleValueType(clazz);
+      return PinotDataType.valueOf(targetType.toUpperCase()).convert(value, sourceType);
```

Comment:
We should make the behavior the same as `CastTransformFunction` (support `INT` and `VARCHAR`). We can make a switch case here for the target data type similar to how we handle the types in `CastTransformFunction`

Reply:
I've done something similar. Please take another look. It needs to be a little more complicated than blindly delegating to PinotDataType in every case.